File: LeetCode/2894_Divisible_and_Non-divisible_Sums_Difference/solution.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function(n, m) {
    let a=n*(n+1)/2;
    let b=Math.floor(n/m);
    b=(b+1)*b*m; // here used ap =>  (n/2(2a+(n-1)d))*2 last 2 was to subtract that number twice, a=0, n=b+1;
    return a-b;
};

File: LeetCode/2894_Divisible_and_Non-divisible_Sums_Difference/solution.cpp
class Solution {
public:
    int differenceOfSums(int n, int m) {
        // int a = n * (n + 1) / 2;
        // int b = m * (n / m) * ((n / m) + 1); 
        // b=(b+1)*b*m;
        // here used ap =>  (n/2(2a+(n-1)d))*2 last 2 was to subtract that number twice, a=0, n=b+1;
        // return a - b;

        return (n * (n + 1) / 2) - (m * (n / m) * ((n / m) + 1));
    }
};